src/Sulu/Bundle/AdminBundle/Controller/AdminController.php

```diff
@@ -34,44 +112,38 @@ class AdminController extends Controller
      */
     public function indexAction()
     {
-        // get user data
-        $userDataServiceId = $this->container->getParameter(self::USER_DATA_ID);
-
-        $user = array();
-        if ($this->has($userDataServiceId)) {
-            /** @var UserManagerInterface $userManager */
-            $userManager = $this->get($userDataServiceId);
-            if ($userManager->getCurrentUserData()->isLoggedIn()) {
-                $user = $userManager->getCurrentUserData()->toArray();
-
-                // get js config from bundles
-                $jsconfig = array();
-                if ($this->has(self::JS_CONFIG_ID)) {
-                    $jsconfig = $this->get(self::JS_CONFIG_ID);
-                    $jsconfig = $jsconfig->getConfigParams();
-                }
-
-                // render template
-                if ($this->get('kernel')->getEnvironment() === 'dev') {
-                    $template = 'SuluAdminBundle:Admin:index.html.twig';
-                } else {
-                    $template = 'SuluAdminBundle:Admin:index.html.dist.twig';
-                }
-
-                return $this->render(
-                    $template,
-                    array(
-                        'name' => $this->container->getParameter('sulu_admin.name'),
-                        'locales' => $this->container->getParameter('sulu_core.locales'),
-                        'suluVersion' => $this->container->getParameter('sulu.version'),
-                        'user' => $user,
-                        'config' => $jsconfig,
-                    )
-                );
-            } else {
-                return $this->redirect($this->generateUrl('sulu_admin.login'));
-            }
+        if (!$this->authorizationChecker->isGranted('IS_AUTHENTICATED_REMEMBERED')) {
+            return new RedirectResponse($this->urlGenerator->generate('sulu_admin.login', array()));
         }
+
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        // get js config from bundles
+        $jsConfig = $this->jsConfigPool->getConfigParams();
+
+        // render template
+        if ($this->environment === 'dev') {
+            $template = 'SuluAdminBundle:Admin:index.html.twig';
+        } else {
+            $template = 'SuluAdminBundle:Admin:index.html.dist.twig';
+        }
+
+        return $this->engine->renderResponse(
+            $template,
+            array(
+                'name' => $this->adminName,
+                'locales' => $this->locales,
+                'suluVersion' => $this->suluVersion,
+                'user' => json_decode(
```

Comment:
Couldn't find a better solution ATM... It would be awesome if the serializer would support arrays, but AFAIK it doesn't ...